mmedit/models/backbones/sr_backbones/rdn.py

```diff
@@ -0,0 +1,187 @@
+import torch
+from mmcv.runner import load_checkpoint
+from torch import nn
+
+from mmedit.models.registry import BACKBONES
+from mmedit.utils import get_root_logger
+
+
+class DenseLayer(nn.Module):
+    """Dense layer
+
+    Args:
+        in_channels (int): Channel number of inputs.
+        out_channels (int): Channel number of outputs.
+
+    """
+
+    def __init__(self, in_channels, out_channels):
+        super(DenseLayer, self).__init__()
+        self.conv = nn.Conv2d(
+            in_channels, out_channels, kernel_size=3, padding=3 // 2)
+        self.relu = nn.ReLU(inplace=True)
+
+    def forward(self, x):
+        """Forward function.
+
+        Args:
+            x (Tensor): Input tensor with shape (n, c, h, w).
+
+        Returns:
+            Tensor: Forward results, tensor with shape (n, c, h, w).
+        """
+        return torch.cat([x, self.relu(self.conv(x))], 1)
+
+
+class RDB(nn.Module):
+    """Residual Dense Block of Residual Dense Network
+
+    Args:
+        in_channels (int): Channel number of inputs.
+        out_channels (int): Channel number of outputs.
+    """
+
+    def __init__(self, in_channels, growth_rate, num_layers):
+        super(RDB, self).__init__()
+        self.layers = nn.Sequential(*[
+            DenseLayer(in_channels + growth_rate * i, growth_rate)
+            for i in range(num_layers)
+        ])
+
+        # local feature fusion
+        self.lff = nn.Conv2d(
+            in_channels + growth_rate * num_layers, growth_rate, kernel_size=1)
+
+    def forward(self, x):
+        """Forward function.
+
+        Args:
+            x (Tensor): Input tensor with shape (n, c, h, w).
+
+        Returns:
+            Tensor: Forward results, tensor with shape (n, c, h, w).
+        """
+        return x + self.lff(self.layers(x))  # local residual learning
+
+
+@BACKBONES.register_module()
+class RDN(nn.Module):
+    """RDN model for single image super-resolution.
+
+    Paper: Residual Dense Network for Image Super-Resolution
+    Adapted from:
+        https://github.com/yulunzhang/RDN.git
+        https://github.com/yjn870/RDN-pytorch
+
+    Args:
+        in_channels (int): Channel number of inputs.
+        out_channels (int): Channel number of outputs.
+        mid_channels (int): Channel number of intermediate features.
+            Default: 64.
+        num_blocks (int): Block number in the trunk network. Default: 16.
+        upscale_factor (int): Upsampling factor. Support 2^n and 3.
+            Default: 4.
+        num_layer (int): Layer number in the Residual Dense Block.
+            Default: 8.
+        growth_rate(int): Channels growth in each layer of RDB.
```

Comment:
`rate`, a terminology that is often related to percentage, is not good to describe this number.

How about just `growth`, or `channel_growth` or `growth_num`, etc.

Reply:
picked `channel_growth`.